library/Translate/src/Juvix/FrontendContextualise/ModuleOpen/Environment.hs

```diff
@@ -83,12 +110,72 @@ newtype Context a = Ctx {antiAlias :: ContextAlias a}
   deriving
     (HasThrow "error" Error)
     via MonadError ContextAlias
+  deriving
+    ( HasReader "dispatch" EnvDispatch,
+      HasSource "dispatch" EnvDispatch
+    )
+    via ReaderField "dispatch" ContextAlias
   deriving
     ( HasReader "openMap" OpenMap,
       HasSource "openMap" OpenMap
     )
     via ReaderField "openMap" ContextAlias
 
+--------------------------------------------------------------------------------
+-- Generic Interface Definitions for Environment lookup
+--------------------------------------------------------------------------------
+
+-- | Names encapsulates the idea of looking up all current names in
+-- a context
+class Names a where
+  type NameConstraint a (m :: * -> *) :: Constraint
+  contextNames :: (NameConstraint a m) => NameSymbol.T -> a -> m [Symbol]
+  currentNameSpace' :: NameConstraint a m => a -> m (NameSymbol.T)
+  inCurrentModule' :: NameConstraint a m => NameSymbol.T -> a -> m Bool
```

Comment:
This interface should make it very easy to test `populateModMap` without having any context logic.

Since all calls hit this, we can setup an interface for this that just returns static things, which we can test against.